### PR TITLE
Improve TLS documentation

### PR DESCRIPTION
**Pull request by tiran**
See: https://github.com/python-ldap/python-ldap/issues/55
Signed-off-by: Christian Heimes <cheimes@redhat.com>

### CODE REVIEW

File: Doc/reference/ldap.rst [hunk 1]
```diff
@@ -221,34 +227,153 @@ SASL options
...
+
+      conn = ldap.initialize("ldap://ldap.example")
+      conn.set_option(ldap.OPT_X_TLS_CACERTFILE, '/path/to/ca.pem')
+      conn.set_option(ldap.OPT_X_TLS_NEWCTX, 0)
+      conn.start_tls_s()
+      conn.simple_bind_s(dn, password)
+
 .. py:data:: OPT_X_TLS
 
+   .. deprecated:: 3.0
```
Comment (encukou) [inline Doc/reference/ldap.rst:244]:
```suggestion
   .. deprecated:: 3.3.0
```

File: Doc/reference/ldap.rst [hunk 2]
```diff
@@ -221,34 +227,153 @@ SASL options
...
+
+   :py:const:`OPT_X_TLS_DEMAND`
+      Validate peer cert chain and host name
+
+   :py:const:`OPT_X_TLS_HARD`
+      Same as :py:const:`OPT_X_TLS_DEMAND`
+
 .. py:data:: OPT_X_TLS_TRY
 
+   .. deprecated:: 3.0
```
Comment (encukou) [inline Doc/reference/ldap.rst:369]:
```suggestion
   .. deprecated:: 3.3.0
```

**Comment by tiran** [comment]
> As expected, TLS docs are alphabet soup: TLS, CA, CRL, PEM, BER/DER
> It would be nice to link to some definitions, or at least expand the abbreviations. Do you know of a place these could link to? Or are the abbreviations the best we can do?

I have sorted the attributes by topic and rephrased some of the words (e.g. ASN.1 structure instead of DER/BER and spelled out certificate revocation list).

**Comment by encukou** [comment]
Thank you!

**Pull request merged**